util/mkdef.pl

```diff
@@ -874,6 +874,10 @@ sub do_defs
 			next if(/\#define/);
 
 			# Reduce argument lists to empty ()
+			# get rid of any existing empty argument lists
+			while(/\(\)/) {
+				s/\(\)//gs;
+			}
 			# fold round brackets recursively: (t(*v)(t),t) -> (t{}{},t) -> {}
 			while(/\(.*\)/s) {
 				s/\([^\(\)]+\)/\{\}/gs;
```

Comment:
Re the .a static libraries, I looked at them briefly. After the fix in this PR, libcrypto.dll.a contains ~4600 entries like d004039.o . Presumably .a is correct. (But I should have said 'Expected 1000's of symbols' earlier.)

Apparently source file name gets lost along the way on mingw64. I am used to *NIX, where foo.c results in foo.o showing in the .a file.